applications/dashboard/models/class.usermodel.php

```diff
@@ -4370,22 +4372,24 @@ public function userCache($User, $UserID = null) {
     }
 
     /**
-     * Cache user's roles
+     * Cache a user's roles.
      *
-     * @param type $UserID
-     * @param type $RoleIDs
-     * @return type
+     * @param int $userID The ID of a user to cache roles for.
+     * @param array $roles A collection of role records associated with the user.
+     * @return bool Was the caching operation successful?
      */
-    public function userCacheRoles($UserID, $RoleIDs) {
-        if (is_null($UserID) || !$UserID) {
+    public function userCacheRoles($userID, $roles) {
+        if (is_null($userID) || !$userID) {
```

Comment:
Pretty sure the original intention of this was:

```php
 if ($userID !== 0 && !$userID) {
```

It will still cache the guest roles this way. Not sure though, but as it stands the `is_null` check is meaningless.